test/jdk/java/nio/file/WatchService/Basic.java

```diff
@@ -68,10 +68,15 @@ static void checkExpectedEvent(Iterable<WatchEvent<?>> events,
         WatchEvent<?> event = events.iterator().next();
         System.out.format("got event: type=%s, count=%d, context=%s\n",
             event.kind(), event.count(), event.context());
-        if (event.kind() != expectedKind)
-            throw new RuntimeException("unexpected event");
-        if (!expectedContext.equals(event.context()))
-            throw new RuntimeException("unexpected context");
+        if (event.kind() != expectedKind) {
+            throw new RuntimeException("unexpected event: " + event.kind() +
+                                       " (expected: " + expectedKind + ")");
+        }
+        if (!expectedContext.equals(event.context())) {
+            System.out.format("context: %s (%s) | expected: %s (%s)", event.context(), event.context().getClass(), expectedContext, expectedContext.getClass());
+            throw new RuntimeException("unexpected context: " + event.context() +
```

Comment:
If you are changing the test then please try to keep it consistent with the existing code if possible. In this case, adding a 150+ line is a bit annoying as it will mess up all future side-by-side diffs.

Reply:
Fair comment. Though, I actually wasn't looking to change the test, I just overlooked this when I removed some changes I had made to help me debug an issue. I have now reverted this change to the test as well.